File: src/codeforces/contest1335/ProblemB.java
package codeforces.contest1335;

import java.util.Scanner;

public class ProblemB {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        for (int t = scanner.nextInt(); t > 0; t--) {
            int n = scanner.nextInt();
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            char[] ans = new char[n];
            for (int i = 0; i < n; i++) {
                ans[i] = (char) ('a' + i % b);
            }
            System.out.println(new String(ans));
        }
    }
}

File: src/codeforces/contest580/ProblemB.kt
@file:Suppress("DuplicatedCode")

package codeforces.contest580

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*
import kotlin.math.max

fun main(args: Array<String>) {
    val n = nextInt()
    val d = nextInt()
    val arr = Array(n) { nextInts(2) }
    val ans = solve(arr, d)
    println(ans)
}

fun solve(arr: Array<IntArray>, d: Int): Long {
    arr.sortBy { it[0] }
    var ans = 0L
    var left = 0
    var curSum = 0L
    for (friend in arr) {
        curSum += friend[1]
        while (arr[left][0] + d <= friend[0]) curSum -= arr[left++][1]
        ans = max(ans, curSum)
    }
    return ans
}


private class MyReader(inputStream: InputStream) {
    private val reader = BufferedReader(InputStreamReader(inputStream))
    private var tokenizer: StringTokenizer? = null

    fun next(): String {
        while (true) {
            tokenizer.let {
                if (it == null || !it.hasMoreTokens()) {
                    tokenizer = StringTokenizer(reader.readLine())
                } else {
                    return it.nextToken()
                }
            }
        }
    }
}

private val reader = MyReader(System.`in`)
private fun next() = reader.next()
private fun nextInt() = next().toInt()
private fun nextLong() = next().toLong()
private fun nextDouble() = next().toDouble()
private fun nextStrings(n: Int) = Array(n) { next() }
private fun nextInts(n: Int) = IntArray(n) { nextInt() }
private fun nextLongs(n: Int) = LongArray(n) { nextLong() }
private fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }